source/Octopus.Tentacle/Maintenance/WorkspaceCleaner.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Octopus.Diagnostics;
+using Octopus.Tentacle.Communications;
+using Octopus.Tentacle.Scripts;
+using Octopus.Tentacle.Services.Scripts;
+using Octopus.Time;
+
+namespace Octopus.Tentacle.Maintenance
+{
+    public class WorkspaceCleaner
+    {
+        readonly WorkspaceCleanerConfiguration configuration;
+        readonly IScriptWorkspaceFactory scriptWorkspaceFactory;
+        readonly IClock clock;
+        readonly ISystemLog log;
+
+        readonly ScriptService scriptService;
+        readonly ScriptServiceV2 scriptServiceV2;
+
+        public WorkspaceCleaner(
+            WorkspaceCleanerConfiguration configuration,
+            IScriptWorkspaceFactory scriptWorkspaceFactory,
+            IServiceRegistration serviceRegistration,
+            IClock clock,
+            ISystemLog log)
+        {
+            this.configuration = configuration;
+            this.scriptWorkspaceFactory = scriptWorkspaceFactory;
+            this.clock = clock;
+            this.log = log;
+
+            scriptService = serviceRegistration.GetService<ScriptService>();
+            scriptServiceV2 = serviceRegistration.GetService<ScriptServiceV2>();
+        }
+
+        public async Task Clean(CancellationToken cancellationToken)
+        {
+            var deleteWorkspacesOlderThanDateTimeUtc = clock.GetUtcTime().DateTime.Subtract(configuration.DeleteWorkspacesOlderThanTimeSpan);
+            log.Info($"Cleaning orphaned workspaces older than {deleteWorkspacesOlderThanDateTimeUtc:g}");
```

Comment:
I know we wanted this to be more or less invisible to the user. But it felt wrong to not have some form of logging in a scheduled repeating task.

Reply:
Just confirming I think this should be Verbose

Reply:
Agreed. Done